addons/account/models/account_move.py

```diff
@@ -1321,36 +1321,122 @@ def _compute_needed_terms(self):
+            move.invoice_has_outstanding = False
+        
+        if not moves_to_process:
+            return
 
+        company_ids = moves_to_process.mapped('company_id')
+        
+        pay_term_lines = self.env['account.move.line'].search([
+            ('move_id', 'in', moves_to_process.ids),
+            ('account_id.account_type', 'in', ('asset_receivable', 'liability_payable'))
+        ])
+        
+        pay_term_lines_by_move = {}
+        for line in pay_term_lines:
+            pay_term_lines_by_move.setdefault(line.move_id.id, set()).add(line.account_id.id)
+        
+        bank_journals_by_company = {}
+        for company in company_ids:
+            bank_journals_by_company[company.id] = self.env['account.journal'].search([
+                *self.env['account.journal']._check_company_domain(company),
+                ('type', '=', 'bank')
+            ]).ids
+
+        all_lines = self.env['account.move.line']
+        move_line_mapping = {}
+        
+        for move in moves_to_process:
+            account_ids = list(pay_term_lines_by_move.get(move.id, set()))
+            if not account_ids:
+                continue
+                
+            domain = expression.AND([
+                _get_common_domain(move.commercial_partner_id.id),
+                [('account_id', 'in', account_ids)]
+            ])
+            
+            # Bank line domain
+            bank_domain = expression.AND([
+                _get_common_domain(move.commercial_partner_id.id),
+                [
+                    ('account_id.account_type', 'in', ['asset_cash', 'asset_receivable']),
+                    ('journal_id', 'in', bank_journals_by_company.get(move.company_id.id, [])),
+                    ('partner_id', '!=', False),
+                ]
+            ])
+            
             if move.is_inbound():
-                domain.append(('balance', '<', 0.0))
-                payments_widget_vals['title'] = _('Outstanding credits')
+                domain = expression.OR([
+                    expression.AND([domain, [('balance', '<', 0.0)]]),
+                    bank_domain
+                ])
             else:
-                domain.append(('balance', '>', 0.0))
-                payments_widget_vals['title'] = _('Outstanding debits')
-
-            for line in self.env['account.move.line'].search(domain):
+                domain = expression.AND([domain, [('balance', '>', 0.0)]])
+            
+            lines = self.env['account.move.line'].search(domain)
+            all_lines += lines
+            move_line_mapping[move.id] = lines
+        
+        if not all_lines:
+            return
+        
+        move_names = set(all_lines.mapped('move_name'))
+        move_counts = {
+            rec[0]: rec[1]
+            for rec in self.env['account.move.line']._read_group(
+                [('move_name', 'in', list(move_names)), ('id', 'in', all_lines.ids)],
```

Comment:
('id', 'in', all_lines.ids) needed ? Feels like it's obviously in it since we have the move and if so maybe we just need to get the move direclty instead of all_lines we would have a set of all_moves